### PR TITLE
CNV-43096: Customize disk size instancetype

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Change disk size in diskmodal option.

Only on customize flag true as it cannot be done on VMs already created

## 🎥 Demo

**Before**

Screenshot 2024-07-16 at 14 19 51
Screenshot 2024-07-16 at 14 19 55

**After**

https://github.com/user-attachments/assets/aebd86ed-2fa2-41b7-8222-94828dae2c21

### CODE REVIEW

File: src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/utils/getEditDiskStates.ts [hunk 1]
```diff
@@ -42,12 +44,14 @@ export const getEditDiskStates: UseEditDiskStates = (vm, diskName, vmi) => {
     const volume = volumes?.find(({ name }) => name === diskName);
     const volumeSource = getVolumeSource(volume);
 
-    if (volumeSource === SourceTypes.EPHEMERAL) {
+    if (volumeSource === VolumeTypes.CONTAINER_DISK) {
```
Comment (metalice) [inline src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/utils/getEditDiskStates.ts:47]: ?

File: src/utils/components/DiskModal/DiskModal.tsx [hunk 2]
```diff
@@ -71,7 +72,8 @@ const DiskModal: FC<DiskModalProps> = ({
         }}
         onSubmit={() =>
           handleSubmit((data) => {
-            if (isEditingCreatedDisk) return editVMDisk(vm, initialFormData, data, onSubmit);
+            if (isEditingCreatedDisk || customize)
+              return editVMDisk(vm, initialFormData, data, onSubmit);
```
Comment (avivtur) [inline src/utils/components/DiskModal/DiskModal.tsx:76]: Please select a different name to this function as it's too similar to editDisk

Comment (upalatucci) [reply src/utils/components/DiskModal/DiskModal.tsx]: Fixed

File: src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/utils/getEditDiskStates.ts (hunk 1 above)
Comment (upalatucci) [reply src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/utils/getEditDiskStates.ts]: Resolved

File: src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/DiskRowActions.tsx [hunk 3]
```diff
@@ -68,9 +97,23 @@ const DiskRowActions: FC<DiskRowActionsProps> = ({ obj, onDiskUpdate, vm, vmi })
...
+          headerText={t('Detach disk?')}
+          isOpen={isOpen}
+          obj={vm}
+          onClose={onClose}
+          onSubmit={onCustomizeDeleteDisk}
+          submitBtnText={deleteBtnText}
+          submitBtnVariant={ButtonVariant.danger}
+        >
+          <ConfirmActionMessage action="detach" obj={{ metadata: { name: diskName } }} />
+        </TabModal>
```
Comment (metalice) [inline src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/DiskRowActions.tsx:112]: can export to a component

Comment (upalatucci) [reply src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/DiskRowActions.tsx]: solved

**Pull request merged**